x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/Ccr.java

```diff
@@ -271,11 +273,17 @@ public Ccr(final Settings settings) {
 
     public List<NamedXContentRegistry.Entry> getNamedXContent() {
         return Arrays.asList(
-                // Persistent action requests
-                new NamedXContentRegistry.Entry(PersistentTaskParams.class, new ParseField(ShardFollowTask.NAME),
+                // auto-follow metadata, persisted into the cluster state as XContent
+                new NamedXContentRegistry.Entry(
```

Comment:
I think we should add assertions in the code base that when named xcontent is used in cluster state then it must be registered.

Reply:
I agree. Would you look into this in a follow-up?

Reply:
Yes, I'm looking into it.

Reply:
I've opened #40171